Objects/object.c

```diff
@@ -1745,16 +1743,10 @@ _PyObject_GenericSetAttrWithDict(PyObject *obj, PyObject *name,
     }
   error_check:
     if (res < 0 && PyErr_ExceptionMatches(PyExc_KeyError)) {
-        if (PyType_IsSubtype(tp, &PyType_Type)) {
-            PyErr_Format(PyExc_AttributeError,
-                         "type object '%.50s' has no attribute '%U'",
-                         ((PyTypeObject*)obj)->tp_name, name);
-        }
-        else {
-            PyErr_Format(PyExc_AttributeError,
-                         "'%.100s' object has no attribute '%U'",
-                         tp->tp_name, name);
-        }
+        assert(!PyType_IsSubtype(tp, &PyType_Type));
```

Comment:
I don't understand this assert: why can't `tp` be a `PyType_Type` subtype?

Reply:
This is to make sure that removing the type specific special case here is safe.  I'll actually hoist this up to the top of the function instead of burying it here as we don't want to use this on types.

Reply:
Moving this actually caused this to get hit in ctypes...  But I think there's a good reason to not do this - the type cache isn't properly getting invalidated if using this directly.